libraries/lib-xml/XMLTagHandler.cpp

```diff
@@ -170,39 +170,32 @@ bool XMLValueChecker::IsGoodInt64(const wxString & strInt)
 
 bool XMLTagHandler::ReadXMLTag(const char *tag, const char **attrs)
 {
-   wxArrayString tmp_attrs;
+   mCurrentTagAttributes.clear();
 
    while (*attrs) {
-      const char *s = *attrs++;
-      tmp_attrs.push_back(UTF8CTOWX(s));
-   }
+      const char* name = *attrs++;
+      const char* value = *attrs++;
 
-// JKC: Previously the next line was:
-// const char **out_attrs = NEW char (const char *)[tmp_attrs.size()+1];
-// however MSVC doesn't like the constness in this position, so this is now
-// added by a cast after creating the array of pointers-to-non-const chars.
-   auto out_attrs = std::make_unique<const wxChar *[]>(tmp_attrs.size() + 1);
-   for (size_t i=0; i<tmp_attrs.size(); i++) {
-      out_attrs[i] = tmp_attrs[i];
+      mCurrentTagAttributes.emplace_back(
+         std::string_view(name), XMLAttributeValueView(std::string_view(value)));
    }
-   out_attrs[tmp_attrs.size()] = 0;
 
-   bool result = HandleXMLTag(UTF8CTOWX(tag), out_attrs.get());
+   bool result = HandleXMLTag(tag, mCurrentTagAttributes);
```

Comment:
By eliminating UTF8CTOWX, are you requring that tag strings to be canonicalized into utf-8 already, or requiring them just to be ASCII strings?  They are all ASCII in practice (I just had to  review them all).  It is worth a comment.

You know I like to say things with type disctinctions where I can.  If we were using C++20, I would use std::u8string_view instead of std::string_view.  Not having C++20 and char8_t, what can we do?  Maybe at least make a different type alias to make the disctinction in function declarations, thought we won't have type checking by the compiler.

Reply:
I don't actually think that char8_t implies any checks on encoding. However, we use Expat, and Expat seems to convert the output to UTF8 internally. So specifically for this case, I imply UTF8, and that is what originally implied.

The UTF8 allows to safely compare to ASCII strings given that all the symbols are from the first half of the table. This covers Audacity needs for 100% when working with tags and names. For attribute values - we explicitly convert from UTF8 to wxString where needed.

Reply:
Certainly it's safe when a tag is ASCII, and all of them are.  But I'm asking, are we _requiring_ them to be ASCII now that you remove UTF3CTOWX ?

You may miss my point about char8_t -- it is that, where a string or string_view is required to be utf8 encoding, it would be good to say that with one type, and where required to be ASCII, to say that without another type, and it would be best if the types do not implicitly interconvert, so the compiler will help us catch misuses.

Reply:
>are we requiring them to be ASCII 

We can't even do it, because Expat will always output UTF8 encoded string.

>where a string or string_view is required to be utf8 encoding

The safest assumption is that all std::string are UTF8 encoded to be fair.